src/Microsoft.Diagnostics.Monitoring/ClientEndpointInfoSource.cs

```diff
@@ -15,12 +16,22 @@ internal sealed class ClientEndpointInfoSource : IEndpointInfoSourceInternal
     {
         public Task<IEnumerable<IEndpointInfo>> GetEndpointInfoAsync(CancellationToken token)
         {
+            bool isInDockerContainer = RuntimeInfo.IsInDockerContainer;
```

Comment:
I've moved the self filter from this class up to the DiagnosticServices class, where we are already doing filtering and and default process selection for dotnet-monitor.

Reply:
I've moved the filtering again into its own class named FilteredEndpointInfoSource in the dotnet-monitor tool project since it is behavior that is specific to dotnet-monitor. It has a two-fold purpose of filtering dotnet-monitor from consideration and selects the IEndpointInfoSource implementation based on the the configuration options provided to it by dependency injection.